src/Elewant/AppBundle/Controller/ShepherdController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Elewant\AppBundle\Controller;
+
+use Elewant\AppBundle\Entity\Herd;
+use Elewant\AppBundle\Repository\HerdRepository;
+use Elewant\UserBundle\Entity\User;
+use Elewant\UserBundle\Security\UserProvider;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * @Route("/shepherd")
+ */
+class ShepherdController extends Controller
+{
+
+    /**
+     * @Route("/{username}", name="shepherd_admire_herd")
+     */
+    public function shepherdAdmireHerdAction($username)
+    {
+        try {
+            $user = $this->getUserByName($username);
+        } catch (UsernameNotFoundException $e) {
+            throw $this->createNotFoundException('The user does not exist');
```

Comment:
Good question.. This specific 404 was due to the fact that the {username} comes from the URL, meaning the request is easily changed by the user and we should prepare for that.

If an existing user does not have a herd, that would be a mistake on our end, more in the 500 range. But I wasn't taking those into account yet... We should. (Tests should catch this)

I think it's clean to have the internal methods throw the exceptions for now, I'll move it.